pkg/reconciler/gc/v2/gc.go

```diff
@@ -65,33 +65,36 @@ func Collect(
 		return a.Before(b)
 	})
 
-	// Delete stale revisions while more than min remain
-	remaining := len(revs)
-	nonstale := make([]*v1.Revision, 0, remaining)
-	for _, rev := range revs {
+	// Delete stale revisions while more than min remain, swap nonstale revisions to the end
+	swap := len(revs)
+	for i := 0; i < swap; {
+		rev := revs[i]
 		switch {
-		case remaining <= min:
+		case len(revs)-i <= min:
 			return nil
 
 		case !isRevisionStale(cfg, rev, logger):
-			nonstale = append(nonstale, rev)
+			swap--
+			revs[i], revs[swap] = revs[swap], revs[i]
```

Comment:
you actually don't need to swap. Just copy tail to the index, since you're discarding the tail anyway.

Reply:
actually in this case we're only keeping the tail and discarding the beginning, the tail maintains the sort order, but in reverse.

This now reads like a programming puzzle imo

Reply:
😁